api/apps/api/src/modules/scenarios/protected-area/protected-area.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { Queue } from 'bullmq';
+import { Either, left, right } from 'fp-ts/Either';
+
+import { JobInput, JobOutput } from '@marxan/protected-areas';
+import { ApiEventsService } from '@marxan-api/modules/api-events';
+
+import { scenarioProtectedAreaQueueToken } from './queue.providers';
+import { API_EVENT_KINDS } from '@marxan/api-events';
+
+export const submissionFailed = Symbol(
+  `System could not submit the async job.`,
+);
+
+@Injectable()
+export class ProtectedAreaService {
+  constructor(
+    @Inject(scenarioProtectedAreaQueueToken)
+    private readonly queue: Queue<JobInput, JobOutput>,
+    private readonly apiEvents: ApiEventsService,
+  ) {}
+
+  async addShapeFor(
+    projectId: string,
+    scenarioId: string,
+    shapefile: JobInput['shapefile'],
+  ): Promise<Either<typeof submissionFailed, true>> {
+    const job = await this.queue.add(`add-protected-area`, {
+      projectId,
+      scenarioId,
+      shapefile,
+    });
+
+    // bad typing? may happen that job wasn't added
```

Comment:
@Sikora00 of the `@types`. Afair if job couldn't be submitted, it can return `undefined` instead a job instance but types says it always return a `Job`

Reply:
actually Kamil is right
see:
https://github.com/taskforcesh/bullmq/blob/master/src/classes/queue.ts#L95
and
https://github.com/taskforcesh/bullmq/blob/8dc7f36fa9e91f1d9f876db82dccbdbd2b8e3bbd/src/classes/repeat.ts#L24
so it is possible to have undefined here where a job is a repeatable job (and probably only in this case)